src/nvim/syntax.c

```diff
@@ -7262,6 +7265,28 @@ int syn_get_final_id(int hl_id)
   return hl_id;
 }
 
+/* Refresh the color attributes of all highlight
+ * groups.
+ * This usually needs to be done after the "Normal"
+ * group is modified to update the groups that might
+ * be using "bg" or "fg".
+ */
```

Comment:
Changing the legacy style is an ongoing process, so old code often still has the old style. For that reason the linter encourages touching adjacent lines (it "cascades" to the next/previous line).